### DISCUSSION TITLE
Is there described architecture somewhere?

### DISCUSSION THREAD

**Discussion by anatoliilitvinovr** [TanStack/table#5532] [General] [answered]
Thanks, community for the great library!

General docs describe code API, but I'm interested in architectural points.
To understand how it works, I'm searching for something like c4 models, UML diagrams, etc.

Thanks for help!

**Comment by KevinVandy** [comment, marked as answer]
This library isn't like a full framework. We have docs and examples. We have very good Typescript types and js doc comments. The closest thing to architectural docs we have is the [custom features guide](https://tanstack.com/table/latest/docs/guide/custom-features), which shows how all the feature code is organized. Also the [Row Models Guide](https://tanstack.com/table/latest/docs/guide/row-models) explains in what order features are applied to data.

**Reply by anatoliilitvinovr** [reply]
That's similar to what I need. Thanks for the clarification!
